pkg/network/setup/netpod/netpod.go

```diff
@@ -106,6 +107,11 @@ func NewNetPod(vmiNetworks []v1.Network, vmiIfaces []v1.Interface, vmiUID string
 	for _, opt := range opts {
 		opt(&n)
 	}
+
+	ifaceStatusesByName := vmispec.IndexInterfaceStatusByName(n.vmiIfaceStatuses, vmispec.HasIfaceStatusOriginatedFromSpec)
+
+	n.podIfaceNamesByNetName = namescheme.CreateFromIfaceStatuses(n.vmiSpecNets, ifaceStatusesByName)
```

Comment:
I commented about it previously, I would move this logic to the caller keeping this factory simple as possible.
Pass the crafted  namescheme from ifaces status via opt. function.

Reply:
So I suggest we will continue the discussion on the previous [thread](https://github.com/kubevirt/kubevirt/pull/13265#discussion_r1869482040).

Reply:
This thread can be resolved.